### PR TITLE
Auditing for specific namespace (with -n or --namespace option)

### PR DESCRIPTION
**Pull request by jerr**
Limiting auditing to a namespace can be useful in some cases.
I propose to add the `-n` or `--namespace` option to do this.

```
kubeaudit -l sat -n default
ERRO[0000] Default serviceAccount with token mounted. Please set AutomountServiceAccountToken to false  KubeType=deployment Name=redis Namespace=default
```
If nothing is indicated, all namespaces are taken into account as currently.

What do you think about this option?

### CODE REVIEW

**Comment by jinankjain** [comment]
Can you add some test for this in the test suite?

**Comment by jerr** [comment]
Hi @jinankjain 
I'm adding tests to get better code coverage but it's not ready yet.
Let me come back to that.

**Comment by jerr** [comment]
I added some namespace tests.
To put it simply, if the error code is negative then it must not appear in the audit list of detected errors.

Let me know if that suits you.

File: cmd/test_util.go [hunk 1]
```diff
@@ -58,5 +59,10 @@ func runTest(t *testing.T, file string, function interface{}, errCode int, argSt
 			errors = append(errors, occurrence.id)
 		}
 	}
-	assert.Contains(errors, errCode)
+	if errCode > 0 {
+		assert.Contains(errors, errCode)
+	} else {
+		assert.NotContains(errors, -errCode)
```
Comment (jinankjain) [inline cmd/test_util.go]: Isn't error codes always positive as they are part of that pesudo enum?

File: cmd/util.go [hunk 2]
```diff
@@ -22,44 +24,53 @@ func debugPrint() {
...
 }
 
+func isInNamespace(meta metav1.ObjectMeta) (valid bool) {
+	return rootConfig.namespace == apiv1.NamespaceAll || rootConfig.namespace == meta.Namespace
+}
+
 func convertDeploymentToDeploymentList(deployment Deployment) (deploymentList *DeploymentList) {
-	deploymentList = &DeploymentList{
-		Items: []Deployment{deployment},
+	if isInNamespace(deployment.ObjectMeta) {
```
Comment (jinankjain) [inline cmd/util.go]: I have a better suggestion for these `convertBla` should return types like `kubeAuditBla` instead of k8s types. This will help in simplifying the code. As later in the code we use convert function to construct those types. Other suggestion would be just get rid of those custom types like `kubeAuditBla` and only use native types from k8s, which would require a lot of refactoring but possible with one type like interface{}[].

**Comment by jerr** [comment]
I changed the tests a bit of namespace_test.go for less impacted existing code (especially in test_util.go).
I hope it's simpler like that.

I'm thinking of improving the runTest function to be able to address more scenarios but it's better to do it in another pull request (like to validate the other data of the results).

File: cmd/util.go (hunk 2 above)
Comment (jerr) [reply cmd/util.go]: I looked at your second suggestion and it seems possible and simplifies the code. Maybe you can create a refactoring issue for that.

File: cmd/namespace_test.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+func TestReplicationControllerInNamespace(t *testing.T) {
+	runTestInNamespace(t, "fakeReplicationControllerASAT", "service_account_token_nil_and_no_name.yml", auditAutomountServiceAccountToken, ErrorServiceAccountTokenNILAndNoName)
+}
+
+func TestReplicationControllerNotInNamespace(t *testing.T) {
+	runTestInNamespace(t, "otherFakeReplicationControllerASAT", "service_account_token_nil_and_no_name.yml", auditAutomountServiceAccountToken)
+}
+
+func runTestInNamespace(t *testing.T, namespace string, file string, function interface{}, errCode ...int) {
```
Comment (klautcomputing) [inline cmd/namespace_test.go]: This should be in `test_util.go`

File: cmd/test_util.go [hunk 4]
```diff
@@ -58,5 +58,9 @@ func runTest(t *testing.T, file string, function interface{}, errCode int, argSt
 			errors = append(errors, occurrence.id)
 		}
 	}
-	assert.Contains(errors, errCode)
+
+	if errCode != 0 {
```
Comment (klautcomputing) [inline cmd/test_util.go]: Good catch 👍  our errors starts at 1

File: cmd/util.go [hunk 5]
```diff
@@ -22,44 +24,53 @@ func debugPrint() {
 	}
 }
 
+func isInNamespace(meta metav1.ObjectMeta) (valid bool) {
```
Comment (klautcomputing) [inline cmd/util.go]: Could we generalize this function so that it takes a namespace as an argument? Because from reading the function name I expected it to take a namespace as an argument.

And then maybe have a second function `isInRootConfigNamespace` (not the best name, but I guess you get the idea) that calls just `isInNamespace(rootConfig.namespace)`

TLDR & easy solution: rename the function :)

File: cmd/util.go (hunk 2 above)
Comment (klautcomputing) [reply cmd/util.go]: I am in favor of merging this imperfect solution and address the underlying and deeper refactoring issue in another (and way way longer) PR.

File: cmd/util.go (hunk 5 above)
Comment (jerr) [reply cmd/util.go]: I have no better proposal for the moment so I'll take yours.

**Comment by klautcomputing** [comment]
I merged this this and we need to address the concern that @jinankjain brought up (that the code gets uglier) in a separate (set of) PR(s)
